File: src/main/java/de/twometer/orion/gl/Cubemap.java
package de.twometer.orion.gl;

import static org.lwjgl.opengl.GL11.glBindTexture;
import static org.lwjgl.opengl.GL13.GL_TEXTURE_CUBE_MAP;

public class Cubemap {

    private final int textureId;

    public Cubemap(int textureId) {
        this.textureId = textureId;
    }

    public void bind() {
        glBindTexture(GL_TEXTURE_CUBE_MAP, textureId);
    }

    public static void unbind() {
        glBindTexture(GL_TEXTURE_CUBE_MAP, 0);
    }

}

File: neko-engine/src/main/kotlin/de/twometer/neko/render/Cubemap.kt
package de.twometer.neko.render

import org.lwjgl.opengl.GL13.*

data class Cubemap(val textureId: Int) {

    fun bind(unit: Int = 0) {
        glActiveTexture(GL_TEXTURE0 + unit)
        glBindTexture(GL_TEXTURE_CUBE_MAP, textureId)
    }

    fun unbind() = glBindTexture(GL_TEXTURE_CUBE_MAP, 0)

}